In [1]:
import numpy as np
import pandas as pd
import sqlite3
import matplotlib.pyplot as plt

path = r"C:\Users\patri\data-analyst-interview\data\artists.db"

In [2]:
conn = sqlite3.connect(path)

tables = pd.read_sql("""SELECT *
                        FROM sqlite_master
                        WHERE type='table';""", conn)
tables

type      name  tbl_name  rootpage  \
0  table   artists   artists         2   
1  table  artworks  artworks       159   

                                                 sql  
0  CREATE TABLE "artists" (\n"Artist ID" INTEGER,...  
1  CREATE TABLE "artworks" (\n"Artwork ID" INTEGE...

In [3]:
artists = pd.read_sql("""SELECT *
                        FROM artists;""", conn)
artists["Artist ID"] = artists["Artist ID"].astype("str")

In [4]:
artwork = pd.read_sql("""SELECT *
                        FROM artworks;""", conn)
artwork["Artist ID"] = artwork["Artist ID"].astype("str")
artwork["Artwork ID"] = artwork["Artwork ID"].astype("str")

# Question 1

In [5]:
pd.read_sql("""SELECT Name, MAX(`Death Year` - `Birth Year`)
                        FROM artists;""", conn)

Name  \
0  Union Paper Bag Machine Company, Philadelphia, PA   

   MAX(`Death Year` - `Birth Year`)  
0                               130

#### That isn't a real artist, so here is the top 5, so we can get some actual names

In [6]:
pd.read_sql("""SELECT Name, `Death Year` - `Birth Year`
                        FROM artists
                        ORDER BY `Death Year` - `Birth Year` DESC
                        LIMIT 5;""", conn)

Name  \
0  Union Paper Bag Machine Company, Philadelphia, PA   
1                                      Leni Matthaei   
2                                          B. Efimov   
3                                 Manoel de Oliveira   
4                                          Hans Erni   

   `Death Year` - `Birth Year`  
0                          130  
1                          108  
2                          108  
3                          107  
4                          106

#### Leni Matthaei has made the most artworks on display at MoMA

# Question 2

In [7]:
pd.read_sql("""SELECT artists.Name, COUNT(1) AS Frequency
                        FROM artists
                        INNER JOIN artworks ON artists.`Artist ID` = artworks.`Artist ID`
                        GROUP BY 1
                        ORDER BY 2 DESC
                        LIMIT 10;""", conn)

Name  Frequency
0              Eugène Atget       5050
1          Louise Bourgeois       3318
2  Ludwig Mies van der Rohe       2566
3      Unknown photographer       1575
4             Jean Dubuffet       1435
5           Lee Friedlander       1317
6             Pablo Picasso       1310
7              Marc Chagall       1162
8             Henri Matisse       1063
9            Pierre Bonnard        894

# Question 3

Here I think it's a good idea to not only check for dimmensions of a rectangle area but also the circle's surface area, so using the formula

$ A = L\cdot w + \pi \cdot r^2 $

In [8]:
pd.read_sql("""SELECT artists.Name, SUM(`Height (cm)`*`Width (cm)`) + SUM((`Diameter (cm)`/2)*(`Diameter (cm)`/2)*3.14) AS `Surface Area`
                        FROM artists
                        INNER JOIN artworks ON artists.`Artist ID` = artworks.`Artist ID`
                        GROUP BY 1
                        ORDER BY 2 DESC
                        LIMIT 10""", conn)

Name  Surface Area
0           James Lee Byars  1.040749e+07
1  Ludwig Mies van der Rohe  7.582402e+06
2          Louise Bourgeois  4.441873e+06
3        Frank Lloyd Wright  2.967754e+06
4       Robert Rauschenberg  2.770040e+06
5             Henri Matisse  1.736799e+06
6             Jean Dubuffet  1.563943e+06
7          Alexander Calder  1.556715e+06
8              Frank Stella  1.329153e+06
9                Kiki Smith  9.676348e+05

# Question 4

In [9]:
innermerge = pd.merge(artists, artwork, on='Artist ID', how='inner')
innermerge.head()

Artist ID          Name_x Nationality Gender  Birth Year  Death Year  \
0         1  Robert Arneson    American   Male      1930.0      1992.0   
1         1  Robert Arneson    American   Male      1930.0      1992.0   
2         2  Doroteo Arnaiz     Spanish   Male      1936.0         NaN   
3         3     Bill Arnold    American   Male      1941.0         NaN   
4         3     Bill Arnold    American   Male      1941.0         NaN   

  Artwork ID                Title          Name_y  Date  ... Classification  \
0      33599  Study for Head Bath  Robert Arneson  1977  ...        Drawing   
1      64139         General Nuke  Robert Arneson  1986  ...          Print   
2      61629           BAS-RELIEF  Doroteo Arnaiz  None  ...          Print   
3      45972     Honey under Sink     Bill Arnold  1971  ...     Photograph   
4      45997    Honey under Chair     Bill Arnold  1971  ...     Photograph   

  Object Number Diameter (cm) Circumference (cm) Height (cm) Length (cm)  \
0       67.1981           NaN                NaN       105.7         NaN   
1      192.1997           NaN                NaN        80.5         NaN   
2       87.1965           NaN                NaN        54.7         NaN   
3       87.1972           NaN                NaN        38.1         NaN   
4       88.1972           NaN                NaN        38.1         NaN   

  Width (cm) Depth (cm)  Weight (kg)  Duration (s)  
0       75.9        NaN          NaN           NaN  
1       59.2        NaN          NaN           NaN  
2       39.7        NaN          NaN           NaN  
3       58.2        NaN          NaN           NaN  
4       57.8        NaN          NaN           NaN  

[5 rows x 26 columns]

In [10]:
#We slice it to only get year
innermerge["AcquiYear"] = innermerge["Acquisition Date"].str.slice(0,4)
innermerge["AcquiYear"] = innermerge["AcquiYear"].astype("float")
lifetime_sellers = innermerge.loc[(innermerge["AcquiYear"] < innermerge["Death Year"]) &
                                  (innermerge["AcquiYear"] > innermerge["Birth Year"]), "Name_x"].unique()
lifetime_sellers

array(['Robert Arneson', 'Jean (Hans) Arp', 'J. Arrelano Fischer', ...,
       'Munetsugu Satomi', 'Ambrogio Pozzi', 'Egidio Bonfante'],
      dtype=object)

In [11]:
len(lifetime_sellers)

2538

#### MoMA acquired artwork from 2538 unique artists in their lifetime

# Question 5

In [12]:
artists.describe()

Birth Year   Death Year
count  11237.000000  4579.000000
mean    1930.852719  1974.287399
std       34.531997    31.153665
min     1730.000000  1795.000000
25%     1910.000000  1958.000000
50%     1936.000000  1980.000000
75%     1956.000000  1998.000000
max     2012.000000  2017.000000

#### About 3000 missing nationalities
#### About 3000 missing genders
#### About 4000 missing birth years
#### About 10000 missing death years, probably because not dead yet

In [13]:
artists.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15091 entries, 0 to 15090
Data columns (total 6 columns):
Artist ID      15091 non-null object
Name           15091 non-null object
Nationality    12603 non-null object
Gender         12019 non-null object
Birth Year     11237 non-null float64
Death Year     4579 non-null float64
dtypes: float64(2), object(4)
memory usage: 707.5+ KB


In [14]:
artists["Artist ID"].str.isnumeric().value_counts()

True    15091
Name: Artist ID, dtype: int64

#### Good news is there's no repeating id

In [15]:
len(artists["Artist ID"].unique())

15091

#### 255x Nationality unknown instead of NaN
#### 2x Nationality Unknown instead of NaN
#### 2x nationality unknown instead of NaN
#### 2x Various, not specific

In [16]:
with pd.option_context('display.max_rows', None, 'display.max_columns', None):
    print(artists["Nationality"].value_counts())

American               5198
German                  930
French                  839
British                 835
Italian                 531
Japanese                498
Swiss                   280
Dutch                   265
Nationality unknown     255
Austrian                243
Canadian                196
Russian                 188
Brazilian               155
Spanish                 153
Argentine               139
Swedish                 130
Mexican                 128
Polish                  125
Danish                  119
Belgian                  89
Czech                    83
Chinese                  81
Israeli                  75
South African            69
Chilean                  61
Finnish                  60
Cuban                    58
Australian               56
Hungarian                52
Norwegian                49
Colombian                42
Venezuelan               41
Korean                   35
Peruvian                 34
Scottish                 28
Indian              

#### 6 of them are written as male instead of Male

In [17]:
artists["Gender"].value_counts()

Male      9820
Female    2193
male         6
Name: Gender, dtype: int64

In [18]:
artwork.describe()

Diameter (cm)  Circumference (cm)    Height (cm)  Length (cm)  \
count    1399.000000           10.000000  111893.000000   736.000000   
mean       23.248939           44.868020      37.712992    89.117417   
std        45.460079           28.631604      48.151347   329.717487   
min         0.635000            9.900000       0.000000     0.000000   
25%         7.900000           23.500000      18.100000    17.031875   
50%        13.700000           36.000000      27.940056    26.700000   
75%        24.782500           71.125000      44.450100    79.100000   
max       914.400000           83.800000    9140.000000  8321.056600   

          Width (cm)    Depth (cm)    Weight (kg)  Duration (s)  
count  111003.000000  11443.000000     298.000000  3.084000e+03  
mean       38.176838     18.291359    1248.278691  7.830060e+03  
std        67.250118     57.703925   11856.456824  1.185046e+05  
min         0.000000      0.000000       0.090000  0.000000e+00  
25%        17.800000      0.000000       5.443200  2.100000e+02  
50%        25.400100      0.700000      19.051100  7.200000e+02  
75%        44.800000     13.335013      65.318000  3.613250e+03  
max      9144.000000   1808.483617  185067.585957  6.283065e+06

#### 50 missing titles
#### 1000 missing Titles
#### 2000 missing Names
#### 2000 missing Dates
#### 20000 missing Medium
#### The dimensions, diameter, circumference, height, length, width, depth, weight, duration is normal to have NaNs because it
#### doesn't have to match all criteria depending on the medium.
#### 6000 missing Acquisition dates

In [19]:
artwork.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 130262 entries, 0 to 130261
Data columns (total 21 columns):
Artwork ID            130262 non-null object
Title                 130210 non-null object
Artist ID             130262 non-null object
Name                  128802 non-null object
Date                  127954 non-null object
Medium                118343 non-null object
Dimensions            118799 non-null object
Acquisition Date      124799 non-null object
Credit                127192 non-null object
Catalogue             130262 non-null object
Department            130262 non-null object
Classification        130262 non-null object
Object Number         130262 non-null object
Diameter (cm)         1399 non-null float64
Circumference (cm)    10 non-null float64
Height (cm)           111893 non-null float64
Length (cm)           736 non-null float64
Width (cm)            111003 non-null float64
Depth (cm)            11443 non-null float64
Weight (kg)           298 non-null flo

In [20]:
artwork["Artist ID"].str.isnumeric().value_counts()

True     122373
False      7889
Name: Artist ID, dtype: int64

#### Some artworks were done by multiple people, or aren't even referenced back to an Artist
#### Important to note that I might then need to clean up this data to get more accurate answers for Question 2 or 3

In [21]:
artwork.loc[~artwork["Artist ID"].str.isnumeric(), "Artist ID"]

65                          6969, 8134
66                          6956, 6957
76                    7661, 8131, 8180
107                          27, 24452
110                   8102, 6703, 4312
111                   8102, 6703, 4312
112                   8102, 4312, 6703
113                   8102, 6703, 4312
151                   8707, 6951, 6952
154             6956, 7559, 7558, 6957
160                   8213, 6132, 8214
161             8213, 6132, 8214, 8216
162             8215, 6132, 8214, 8216
163             8215, 6132, 8214, 8216
164             8215, 6132, 8214, 8216
178                         6076, 8112
195                         6934, 8103
196                         6934, 8103
238                        8218, 22884
239                        8218, 22884
240                        8218, 22884
241                 29711, 8218, 22884
242             6956, 7558, 7559, 6957
253                          27, 24452
254                   3381, 8157, 8158
266                   338

In [22]:
artwork["Artwork ID"].str.isnumeric().value_counts()

True    130262
Name: Artwork ID, dtype: int64

#### Good thing there's no repeating index

In [23]:
len(artwork["Artwork ID"].unique())

130262

#### We have forks on dates instead of specific years, so the format isn't even consistent

In [24]:
artwork.loc[:20, "Date"]

0        1896
1        1987
2        1903
3        1980
4        1903
5     1976-77
6     1976-77
7     1976-77
8     1976-77
9     1976-77
10    1976-77
11    1976-77
12    1976-77
13    1976-77
14    1976-77
15    1976-77
16    1976-77
17    1976-77
18    1976-77
19    1976-77
20    1976-77
Name: Date, dtype: object

#### The date format is not consistent. Sometimes its trying to fork using the last 2 digits sometimes it actually puts the year

In [25]:
artwork[artwork["Date"].str.match("[0-9]+-[0-9]{4}", na = False)]

Artwork ID                                              Title  \
789           879  Generator Project, White Oak, Florida, Design ...   
791           881  Generator Project, White Oak, Florida, Investi...   
792           882  Generator Project, White Oak, Florida, Perpsec...   
797           887    Generator Project, White Oak, Florida, Sketches   
849           940  Furniture Manufacturers Association Headquarte...   
882           991  Chikatsa-Asuka Historical Museum, Minami-Kawac...   
892          1002                      Cité de Refuge, Paris, France   
917          1029               Glass House, New Canaan, Connecticut   
920          1032    Vulcania, St. Ours-les-Roches, Auvergne, France   
924          1036  Berlin Museum with the Jewish Museum, Scale mo...   
926          1038  Federal Building and United States Courthouse,...   
1171         1777                Sample Blanket Showing Three Weaves   
1406         2113                                              Ladle   
1426         2141                                       Prayer Bench   
1460         2181                                            Tumbler   
1465         2186                                         Wine Glass   
1505         2230                                               Bowl   
1511         2237                                               Bowl   
1542         2271                                              Woman   
1561         2292                                            Pitcher   
1613         2353                                  Toothbrush Holder   
1669         2417                        Group of 18 Textile Samples   
1797         2557                                      Mug and Plate   
1848         2616                                      Schnapps Cups   
1864         2636          Floor Tiles from the Casa Mila, Barcelona   
1871         2644                                               Tray   
1878         2652                               Egg Cup and Salt Cup   
1908         2690                                             Plates   
2039         2844                             Flint Arrowhead Knives   
2084         2893                           Triennale Cocktail Glass   
...           ...                                                ...   
128064     202124                               Werbe-Bau letterhead   
128065     202125                               Werbe-Bau letterhead   
128066     202126                               Werbe-Bau letterhead   
128246     202900     Max Headroom: original scripts and storyboards   
128297     202992                               Location Proposal #2   
128355     203240  W. Sandberg letterhead (Letter to Arthur A. Co...   
128356     203241                             W. Sandberg letterhead   
128482     203703                                               Life   
128679     204622         Solomon R. Guggenheim Museum, New York, NY   
128684     204647                                The Sky's the Limit   
128685     204648                                 The Upward Journey   
128686     204649                                 The Upward Journey   
128688     204651                                   The Three Graces   
128697     204660                               Untitled (Wide Tree)   
128809     204830                                       Child's Play   
129015     205576                               Location Proposal #2   
129066     205830  Herbert Greenwald Apartment Project, Chicago, ...   
129310     207249   H.C. Price Company Tower, Bartlesville, Oklahoma   
129311     207252   Beth Sholom Synagogue, Elkins Park, Pennsylvania   
129312     207255  Marin County Civic Center, San Rafael, California   
129381     214727  Commonwealth Promenade Apartment Buildings, Ch...   
129382     214728  Ron Bacardi y Compania, S.A., Administration B...   
129497     215479                                          Shattered   
129501     215483                              

#### Some even have parenthesis for the fork

In [26]:
artwork[artwork["Date"].str.match("\([0-9]+-[0-9]{4}\)", na = False)]

Artwork ID                                              Title  \
31790       33280            Rocks Near the Caves above Château Noir   
32889       34536   Set design for Der Blaue Cabaret Theatre, Berlin   
34082       36091              How Briefly - The Tale of Two Meeting   
34083       36092              How Briefly - The Tale of Two Meeting   
34161       36183                                     Background #23   
35457       37799                                         Still Life   
35747       38203                                Portrait of a Woman   
36186       38708                                           Untitled   
36187       38709                                           Untitled   
56472       60093                       Three-Sided Elevation on Red   
56931       60583  Rock (Membership Card for the New Artists' Ass...   
57013       60673                       4 Lifestyles (4 Lebensarten)   
57110       60773                                             Bastos   
57152       60816                                         Samurai II   
57242       60918                                      SELF PORTRAIT   
57428       61116                                           FESTIVAL   
57566       61267                                          Car Crash   
57718       61434                         Garden Gate (Gartenpforte)   
57990       61747                                               FIRE   
58067       61826                      Study of a Nude (Etude de nu)   
59308       63151                                            OCTOBER   
59311       63154                                 Sheila Over London   
59370       63213                              Sheila Over London II   
59386       63229              Self Portrait Antrim Hotel, Liverpool   
59935       63819                                           Untitled   
60497       64463                                    Nine from Folio   
60498       64464                           Colophon page from Folio   
60499       64465                                     One from Folio   
60500       64466                                     Two from Folio   
60501       64467                                   Three from Folio   
...           ...                                                ...   
85018       96690  A Month of Drawings in the Cursive Style (no. 23)   
85019       96691   A Month of Drawings in the Cursive Style (no. 9)   
86053       97807                              Cryptonymous Infusion   
86400       98518                                        The Centaur   
86613       98894                                        Storyboards   
86614       98895                                        Storyboards   
86615       98896                                        Storyboards   
86616       98897                                        Storyboards   
86617       98898                                        Storyboards   
86618       98899                                        Storyboards   
86619       98900                                        Storyboards   
86620       98901                                        Storyboards   
86621       98902                                        Storyboards   
86622       98903                                        Storyboards   
88427      103783  Functional Pictures (Gebrauchsbilder) from Mus...   
90495      108280                            Four Pictures of Daphne   
94081      119345  Untitled, plate 4, third version, state II, fr...   
94082      119346  Untitled, plate 5, third version, state X, fro...   
94083      119347  Untitled, plate 6, fifth version, state II, fr...   
94086      119350  Untitled, plate 9, third version, state II, fr...   
94126      119391  Untitled, plate 4, third version, state II, fr...   
94142      119407  Untitled, plate 5, third version, state X, var...   
94147      119412  Untitled, plate 6, fifth version, state II, fr...   
94372      119640  Untitled, plate 1, third ver

In [27]:
artwork["Medium"].value_counts()

Gelatin silver print                                                                                                                                                                                                                                                                             14103
Lithograph                                                                                                                                                                                                                                                                                        7034
Albumen silver print                                                                                                                                                                                                                                                                              4845
Lithograph, printed in color                                                                                       

In [28]:
artwork["Acquisition Date"].head()

0    1996-04-09
1    1995-01-17
2    1997-01-15
3    1995-01-17
4    1997-01-15
Name: Acquisition Date, dtype: object

In [29]:
artwork["Acquisition Date"].str.match("[0-9]{4}-[0-9]{2}-[0-9]{2}", na = True).value_counts()

True     130255
False         7
Name: Acquisition Date, dtype: int64

#### The acquisition date format isn't consistent either, it usually goes YYYY-MM-DD, but here we can see it only use YYYY-MM
#### Also it could even be DD, we don't know

In [30]:
artwork.loc[~artwork["Acquisition Date"].str.match("[0-9]{4}-[0-9]{2}-[0-9]{2}", na = True), "Acquisition Date"]

130209    2016-10
130210    2016-05
130211    2016-05
130212    2016-10
130213    2016-10
130214    2016-10
130215    2016-05
Name: Acquisition Date, dtype: object

#### One of the acquisition year is from 1216, which makes no sense for the Museum of Modern Art as it was established in 1929

In [31]:
aquiyear = artwork["Acquisition Date"].str.slice(0,4).astype(float)
aquiyear.describe()

count    124799.000000
mean       1983.524572
std          22.593042
min        1216.000000
25%        1965.000000
50%        1983.000000
75%        2006.000000
max        2016.000000
Name: Acquisition Date, dtype: float64

In [32]:
artwork["Credit"].value_counts().head()

The Louis E. Stern Collection                                10927
Purchase                                                      8398
Gift of the artist                                            7191
Abbott-Levy Collection. Partial gift of Shirley C. Burden     4889
The Gilbert and Lila Silverman Fluxus Collection Gift         4603
Name: Credit, dtype: int64

In [33]:
artwork["Catalogue"].value_counts()

Y    71333
N    58929
Name: Catalogue, dtype: int64

In [34]:
artwork["Department"].value_counts()

Prints & Illustrated Books               60128
Photography                              29161
Architecture & Design                    18269
Drawings                                 11027
Painting & Sculpture                      3806
Film                                      3088
Media and Performance Art                 2627
Fluxus Collection                         2135
Architecture & Design - Image Archive       21
Name: Department, dtype: int64

#### Not very clear what they mean by Mies van der Rohe and Frank Lloyd Archive in this column

In [35]:
artwork["Classification"].value_counts()

Print                             30807
Photograph                        29909
Illustrated Book                  26160
Drawing                           11735
Design                            11223
Mies van der Rohe Archive          3331
Architecture                       2947
Video                              2363
Film                               2292
Painting                           2270
Sculpture                          1669
Multiple                           1030
(not assigned)                     1029
Frank Lloyd Wright Archive          785
Periodical                          741
Installation                        596
Work on Paper                       436
Audio                               429
Media                               343
Ephemera                             89
Textile                              33
Performance                          24
Collage                               9
Photography Research/Reference        4
Film (object)                         3


#### I don't know what this is. There seems to be number and codes

In [36]:
artwork["Object Number"]

0            885.1996
1              1.1995
2              1.1997
3              2.1995
4              2.1997
5            3.1995.1
6         3.1995.1-24
7           3.1995.10
8           3.1995.11
9           3.1995.12
10          3.1995.13
11          3.1995.14
12          3.1995.15
13          3.1995.16
14          3.1995.17
15          3.1995.18
16          3.1995.19
17           3.1995.2
18          3.1995.20
19          3.1995.21
20          3.1995.22
21          3.1995.23
22          3.1995.24
23           3.1995.3
24           3.1995.4
25           3.1995.5
26           3.1995.6
27           3.1995.7
28           3.1995.8
29           3.1995.9
             ...     
130232         FC5013
130233         FC5014
130234     FC5015.1-2
130235         FC5016
130236         FC5017
130237         FC5018
130238     FC5019.1-2
130239         FC5020
130240         FC5021
130241         FC5022
130242         FC5023
130243         FC5024
130244         FC5025
130245     FC5026.1-3
130246    

# Question 6

In [37]:
artists.loc[artists["Gender"] == "male", "Gender"] = "Male"
artists["Gender"].value_counts()

Male      9826
Female    2193
Name: Gender, dtype: int64

In [38]:
innermerge = pd.merge(artists, artwork, on='Artist ID', how='inner')

In [39]:
innermergeMen = innermerge[innermerge["Gender"] == "Male"]
innermergeWomen = innermerge[innermerge["Gender"] == "Female"]

#### We can seperate the artworks by gender, we see the MoMA has 6x more art made by men than women

In [40]:
innermergeMen.shape[0]/innermergeWomen.shape[0]

6.174519854258013

In [41]:
innermerge["AcquiYear"] = innermerge["Acquisition Date"].str.slice(0,4).astype(float)

In [42]:
innermerge["AcquiYear"].describe()

count    117243.000000
mean       1982.427045
std          22.448878
min        1216.000000
25%        1964.000000
50%        1980.000000
75%        2005.000000
max        2016.000000
Name: AcquiYear, dtype: float64

#### Cluster it by acquisition dates

In [43]:
EarlyAcquisition = innermerge.loc[innermerge["AcquiYear"] < 1960, :]
MidAcquisition = innermerge.loc[(innermerge["AcquiYear"] >= 1960) & (innermerge["AcquiYear"] < 2000), :]
LateAcquisition = innermerge.loc[innermerge["AcquiYear"] >= 2000, :]

In [44]:
print(EarlyAcquisition.shape)
print(MidAcquisition.shape)
print(LateAcquisition.shape)

(15588, 27)
(63350, 27)
(38305, 27)


#### I can also cluster it based on Date but date has to be extensively cleaned up. Theres quite a lot of missing values and its extremely inconsistent.

#### We can also cluster by classification

In [45]:
innermerge["Classification"].value_counts()

Print                             29786
Photograph                        28400
Illustrated Book                  25326
Drawing                           11509
Design                             9793
Mies van der Rohe Archive          3305
Painting                           2260
Architecture                       2196
Video                              1944
Film                               1882
Sculpture                          1653
Multiple                            961
Frank Lloyd Wright Archive          730
(not assigned)                      723
Installation                        553
Periodical                          444
Media                               283
Audio                               256
Work on Paper                       214
Ephemera                             83
Textile                              32
Performance                          21
Collage                               9
Film (object)                         3
Software                              3


In [46]:
Print = innermerge[innermerge["Classification"] == "Print"]
Photography = innermerge[innermerge["Classification"] == "Photograph"]
Book = innermerge[innermerge["Classification"] == "Illustrated Book"]
Drawing = innermerge[innermerge["Classification"] == "Drawing"]
Paint = innermerge[innermerge["Classification"] == "Painting"]
Sculpture = innermerge[innermerge["Classification"] == "Sculpture"]
#Lets merge the archives together, more research would be needed to see if its appropriate to do so
Archives = innermerge[(innermerge["Classification"] == "Frank Lloyd Wright Archive") |
                      (innermerge["Classification"] == "Mies van der Rohe Archive")]
#We combine all types of media because media could be visual or audio
Media = innermerge[(innermerge["Classification"] == "Video") |
                      (innermerge["Classification"] == "Film") |
                      (innermerge["Classification"] == "Audio") |
                      (innermerge["Classification"] == "Media")]

#They're too small to have their own category so we'll put them as others
not_others = ["Print", "Photograph", "Illustrated Book", "Drawing", "Painting", "Sculpture", "Frank Lloyd Wright Archive",
             "Mies van der Rohe Archive", "Video", "Film", "Audio", "Media"]
Other = innermerge[~(innermerge["Classification"].isin(not_others))]

#### We could classify by department

In [47]:
innermerge["Department"].value_counts()

Prints & Illustrated Books               57748
Photography                              27810
Architecture & Design                    16009
Drawings                                 10865
Painting & Sculpture                      3790
Film                                      2407
Media and Performance Art                 2233
Fluxus Collection                         1493
Architecture & Design - Image Archive       18
Name: Department, dtype: int64

In [48]:
PrintDep = innermerge[innermerge["Department"] == "Prints & Illustrated Books"]
PhotographyDep = innermerge[innermerge["Department"] == "Photography"]
DesignDep = innermerge[innermerge["Department"] == "Architecture & Design"]
DrawingsDep = innermerge[innermerge["Department"] == "Drawings"]
PaintAndSculptureDep = innermerge[innermerge["Department"] == "Painting & Sculpture"]
#Here we can combine Film with Media and Performance Art because they're small enough compared to the other categories
#and similar enough for us to do it
MediaDep = innermerge[(innermerge["Department"] == "Media and Performance Art") | (innermerge["Department"] == "Film") ]
#They're too small to have their own category so we'll put them as others
OtherDep = innermerge[(innermerge["Department"] == "Fluxus Collection") | (innermerge["Department"] == "Architecture & Design - Image Archive")]